libs/ledgerjs/packages/keyring-eth/src/DefaultKeyringEth.ts

```diff
@@ -1,46 +1,120 @@
-import { Transaction } from "ethers";
-import Transport from "@ledgerhq/hw-transport";
+import { Transaction, ethers } from "ethers";
 import AppBinding from "@ledgerhq/hw-app-eth";
 import {
   EcdsaSignature,
   KeyringEth,
   GetAddressOptions,
   SignMessageOptions,
-  SignMessagePayload,
   SignTransactionOptions,
   GetAddressResult,
+  SignMessageMethod,
+  SignMessageType,
 } from "./KeyringEth";
+import { ContextResponse } from "@ledgerhq/context-module";
+import { ContextModule } from "@ledgerhq/context-module/lib/ContextModule";
 
 export class DefaultKeyringEth implements KeyringEth {
   private _appBinding: AppBinding;
+  private _contextModule: ContextModule;
 
-  constructor(transport: Transport) {
-    this._appBinding = new AppBinding(transport);
+  constructor(appBinding: AppBinding, contextModule: ContextModule) {
+    this._appBinding = appBinding;
+    this._contextModule = contextModule;
   }
 
   public async signTransaction(
-    _derivationPath: string,
-    _transaction: Transaction,
-    _options: SignTransactionOptions,
+    derivationPath: string,
+    transaction: Transaction,
+    options: SignTransactionOptions,
   ) {
-    // TODO: implement
-    return Promise.resolve({} as EcdsaSignature);
+    const challenge = await this._appBinding.getChallenge();
+
+    const contexts: ContextResponse[] = await this._contextModule.getContexts(transaction, {
+      challenge,
+      options,
+    });
+
+    for (const context of contexts) {
+      if (context.type === "error") {
+        // TODO: handle error here
+        continue;
+      }
+
+      await this._appBinding[context.type](context.payload);
+    }
+
+    const response = await this._appBinding.signTransaction(
+      derivationPath,
+      ethers.utils.serializeTransaction(transaction),
```

Comment:
```suggestion
      ethers.utils.serializeTransaction(transaction).slice(2),
```
As of today at least, `hw-app-eth` expects a an unprefixed hexadecimal string, so no `0x` and ethers will return on with `0x`